### DISCUSSION TITLE
Add a placeholder for get<string>() is value is nullptr

### DISCUSSION THREAD

**Discussion by alex-lt-kong** [nlohmann/json#4114] [Q&A] [answered]
Currently I am doing something like this
```C++
std::string myVal = myJson["myKey"] == nullptr ? "<empty>": myJson["myKey"].get<string>();
```
to make sure myVal is always a valid `std::string`. Wondering if it is possible to supply a placeholder in case of `null` of the JSON object, perhaps something like this:
```C++
std::string myVal = myJson["myKey"].get<string>(if_null="<empty>");
```

**Comment by gregmarr** [comment, marked as answer]
Yes

    std::string myVal = myJson.value("myKey", "<empty>");

**Comment by alex-lt-kong** [comment]
Just tried the method mentioned by @gregmarr . Works like a charm! In case anyone wonders, if we have a nested value, can achieve the same like below:
```C++
std::string myVal = myJson.value("/myKey/alsoMyKey/moreKey"_json_pointer, "<empty>");
```
